molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/AttributeValidator.java

```diff
@@ -308,57 +309,87 @@ private static void validateOrderBy(Attribute attr, Sort orderBy)
 
 	private static void validateUpdateDataType(AttributeType currentDataType, AttributeType newDataType)
 	{
-		EnumSet<AttributeType> disallowedDataTypes = DATA_TYPE_DISALLOWED_TRANSITIONS.get(currentDataType);
-		if (disallowedDataTypes.contains(newDataType))
+		EnumSet<AttributeType> allowedDatatypes = DATA_TYPE_ALLOWED_TRANSITIONS.get(currentDataType);
+		if (!allowedDatatypes.contains(newDataType))
 		{
 			throw new MolgenisDataException(
 					format("Attribute data type update from [%s] to [%s] not allowed, allowed types are %s",
-							currentDataType.toString(), newDataType.toString(),
-							EnumSet.complementOf(disallowedDataTypes).toString()));
+							currentDataType.toString(), newDataType.toString(), allowedDatatypes.toString()));
 		}
 	}
 
-	private static EnumMap<AttributeType, EnumSet<AttributeType>> DATA_TYPE_DISALLOWED_TRANSITIONS;
+	private static EnumMap<AttributeType, EnumSet<AttributeType>> DATA_TYPE_ALLOWED_TRANSITIONS;
 
 	static
 	{
 		// transitions to EMAIL and HYPERLINK not allowed because existing values not checked by PostgreSQL
 		// transitions to CATEGORICAL_MREF and MREF not allowed because junction tables updated not implemented
```

Comment:
The important thing here it not the comments (which should not refer to PostgreSQL, my bad) but: The transitions should be supported by molgenis-data-validation but blocked by molgenis-data-postgresql until we implement them.